diff -Naur neutrino-mp-cst-next.org/src/driver/vfd.cpp neutrino-mp-cst-next/src/driver/vfd.cpp
--- neutrino-mp-cst-next.org/src/driver/vfd.cpp	2015-03-22 09:05:07.000000000 +0100
+++ neutrino-mp-cst-next/src/driver/vfd.cpp	2015-03-21 13:17:00.663677808 +0100
@@ -151,11 +151,15 @@
 		return;
 	}
 
+#if 0 //Let's always start from begining, like in openPLI
 	if (strlen(str)<VFDLENGTH)
 		ws = (VFDLENGTH-strlen(str))/2;
 	else
 		ws = 0;
-	memset(data.data, ' ', 63);
+#else
+		ws = 0;
+#endif		
+		memset(data.data, ' ', 63);
 	if (!fromScrollThread)
 	{
 		memcpy (data.data+ws, str, VFDLENGTH-ws);
